sympy/functions/special/hyper.py

```diff
@@ -284,7 +284,46 @@ def _eval_nseries(self, x, n, logx, cdir=0):
             den = Mul(*[RisingFactorial(b, i) for b in bq])
             terms.append(((num/den) * (arg**i)) / factorial(i))
 
-        return (Add(*terms) + Order(x**n,x))
+        result = Add(*terms)
+        newn = min((Abs(a) for a in ap if a.is_negative), default=n)
+
+        if newn < n:
+            return result
+        return result + Order(x**n, x)
+
+    def _eval_aseries(self, n, args0, x, logx):
+
+        from sympy.functions.special.gamma_functions import gamma
+        from sympy.series.order import Order
+
+        a, b = self.ap
```

Comment:
len(ap) is not always 2.

Reply:
Thanks for reviewing, I was a bit confused with the hyper module and had  implemented the series expansion for 2F1 at first, I later went through the docs and  it turns out the following is the general [expression](https://functions.wolfram.com/HypergeometricFunctions/HypergeometricPFQ/06/01/05/02/0002/).